### DISCUSSION TITLE
How to enable the Save function?

### DISCUSSION THREAD

**Discussion by dkag123** [winder/Universal-G-Code-Sender#2236] [Help]
I can  get the 'save as' to work (by first selecting the projects Gcode window) but I can never seem to get the 'save' to highlight. 
This will save the project so I can open it again with its progress and XYZ zero points still intact (after a homeing). 

I am looking to do the same but without having to create new names for the project every time I save it, or overwriting would be fine as well.

Thanks

**Comment by breiler** [comment]
I am not sure what you mean. 

The save should only be able to save any changes to the gcode file, doesn't that work for you? 

It can not save the controller state such as the coordinate offsets or which line it was previously executing.

**Reply by dkag123** [reply]
If I:
Home machine
Open a project to run (something I made in carbide create)
Set the XYZ
Select the projects Gcode window, now the 'save as' selection is available.
Rename it and save.

Close and re-open UGS
Home machine
Open the project just saved in UGS
Now I can go to the previously saved XYZ coordinates 

I go to those coordinates using 2 old saved macros (one goes to XY, and one macro goes to Z)
I dont use the ones in UGS only because I want to go to the Z coordinence separately (just in case of a bit crash)

... but I never see the 'save' highlighted, just the 'save as'

**Reply by dkag123** [reply]
Oh, if the 'save' is only for Gcode changes, this may be why I dont see it highlighted... I dont know gcode good enough to make changes in it directly so I never do.

**Reply by dkag123** [reply]
![pic-1480](https://github.com/winder/Universal-G-Code-Sender/assets/58303690/823e060a-b2b0-4004-88f1-84efde0f1952)

This is the window I am referring to.. select it first, then save as..(after the XYZ are set )

**Reply by dkag123** [reply]
If you try this and it dont save correctly...
Something I may have left out when saving, it may be needed but I dont think so...
Before you 'save as' have UGS move the spindle to the XY positions (Z if needed),  then save.

**Reply by breiler** [reply]
It will not save the controller state such as its work position. It will only save changes to the gcode file.

**Reply by dkag123** [reply]
It is for me. I am using 2.0.15
I have tested it several times, each time it goes right to to XY (those are the important ones for me, though I do use Z as well)

**Reply by dkag123** [reply]
if its a bug, please dont fix it .. LOL 
it's a great time saver

**Reply by breiler** [reply]
It is not a bug, because what you describe doesn't exist. =)

The work position is stored in the memory on your controller. So if you run another job and use another work offset setting the previous work position will be lost.

**Reply by dkag123** [reply]
Oh, OK. I think I have may shut off and restarted the computer but never the CNC machine itself during a job. I can only do one job at a time and when its done (taken off the machine), thats it.. its done.

It would be a great feature to have if possible. 

Before I found this 'save as' what I was doing is taking a screen snapshot of the XYZ positions. Then manually move to those XYZ positions and reset them, (before running the job again) if I had to close UGS for some reason
